Blocks/Persistence/file.h

```diff
@@ -200,44 +211,47 @@ class FilePersister {
 
   template <typename E>
   idxts_t DoPublish(E&& entry, const std::chrono::microseconds timestamp) {
-    const idxts_t current(impl_->next_index, timestamp);
-    if (current.us < impl_->next_timestamp.load()) {
-      CURRENT_THROW(InconsistentTimestampException(impl_->next_timestamp, current.us));
+    auto last_published_and_end = impl_->last_published_and_end.load();
+    if (timestamp < last_published_and_end.second.second) {
+      CURRENT_THROW(InconsistentTimestampException(last_published_and_end.second.second, timestamp));
     }
+    last_published_and_end.second.second = timestamp;
+    const auto current = idxts_t(last_published_and_end.second.first, timestamp);
     impl_->appender << JSON(current) << '\t' << JSON(std::forward<E>(entry)) << std::endl;
-    ++impl_->next_index;
-    impl_->next_timestamp = current.us + std::chrono::microseconds(1);
+    last_published_and_end.first = last_published_and_end.second;
+    ++last_published_and_end.second.first;
+    last_published_and_end.second.second += std::chrono::microseconds(1);
+    impl_->last_published_and_end.store(last_published_and_end);
```

Comment:
@mzhurovich indeed, and you've convinced me it's unnecessary.

Let's fall back to the simple logic. Please see https://github.com/c5t/Current/pull/394. I guess I'd put a poor man's atomic back, at least for g++, so that and we could move on with our lives.